searvey/coops.py

```diff
@@ -19,132 +21,55 @@
 from typing import Union
 
 import geopandas
+import httpx
+import limits
+import multifutures
 import numpy
 import pandas
+import pandas as pd
 import requests
 import shapely
+import tenacity
 import xarray
 from bs4 import BeautifulSoup
 from bs4 import element
+from deprecated import deprecated
 from geopandas import GeoDataFrame
 from pandas import DataFrame
 from pandas import Series
 from shapely.geometry import MultiPolygon
-from shapely.geometry import Point
 from shapely.geometry import Polygon
 from xarray import Dataset
 
+from .custom_types import DatetimeLike
 from .utils import get_region
 
 
 logger = logging.getLogger(__name__)
 
+# constants
+COOPS_URL_TS_FORMAT = "%Y%m%d %H:%M"
+COOPS_BASE_URL = "https://tidesandcurrents.noaa.gov/api/datagetter?"
+MAX_COOPS_ITERVAL = 30  # TODO: It should depend on interval chosen
```

Comment:
@pmav99 I have to think more about this one, but the max for COOPS depends on both product and interval. Do you have any suggestion?